### ISSUE TITLE
Not able to verify known good WCS solutions

### ISSUE THREAD

**Issue by EAlexJ** [user29A/fastrometry#5]
Hello Joe, Hello Cameron,
I have spent some time getting to grips with Astrometry and the fastrometry package as my tool of choice. While working on some files of my own I ran into issues, namely fastrometry not being able to produce meaningful results after finding pse sources. The files used are of high enough quality to do meaningful astrometry, yet I did not manage to reproduce result from a different pipeline that was used to come up with a solution. In fact, every file from the source im working on failed.
Judging from the very useful debug outputs included in the package, things go wrong when setting the intermediate points.

You can find one of the files im struggling with [here](https://drive.google.com/file/d/18DwVZWoN1R3ub-1yd7h7w-UiRUJoIc_E/view?usp=sharing).

Any help is appreciated.
-Eyck

**Comment by user29A** [comment]
Hi Eyck,

With the CCDLAB implementation I am able to get this solution:

CD1_1   =    -4.0920544817e-05 / WCS rotation and scaling matrix                
CD1_2   =     3.3299290755e-06 / WCS rotation and scaling matrix                
CD2_1   =    -2.3589855261e-06 / WCS rotation and scaling matrix                
CD2_2   =     -4.261630523e-05 / WCS rotation and scaling matrix                
CDELT1  =           0.14780091 / WCS plate scale on axis 1 (arcsec per pixel)   
CDELT2  =           0.15365356 / WCS plate Scale on axis 2 (arcsec per pixel)   
CROTA1  =           4.65222082 / WCS field rotation angle on axis 1 (degrees)   
CROTA2  =           3.16832068 / WCS field rotation angle on axis 2 (degrees)   

Do those values look OK?

For fastrometry, the problem for these images is likely that the default options settings aren't appropriate, as the stars are quite "large". Trying this additional setting:

-kernelrad 21

That should help the PSE, which is likely why there was a problem with setting the intermediate points. The default kernelrad is only 2, so needing to go to 21 is a huge difference.

Let us know if that helps.

Joe

**Comment by EAlexJ** [comment]
The values look okay, but I am not able to reproduce your results whatsoever. I am checking out CCDLAB at the moment to try and find some values this way but Id like to only rely on the fastrometry pipeline in the end.

**Comment by user29A** [comment]
Yes definitely. I didn't try out variations on fastrometry yet myself - did you try the kernelrad argument?

List of all of the options is here:

https://github.com/user29A/fastrometry/wiki/Usage-and-Options

I think you can also get that list with fastrometry -h

**Comment by cam92473** [comment]
Hi Eyck,

Just wanted to let you know I am having a look at it. There is something
strange going on with the algorithm, I agree. Can you check if you have the
most recent version of fastrometry? (Earlier there was a bug with the
intermediate points that has been resolved). But even with that fixed the
algorithm doesn't work on the image you provided. I also tried flipping the
image, but it still doesn't solve. Can you let me know if updating
fastrometry fixes the issue with the intermediate points you are having?

If I vertically flip the image and use -ra 328.676 -dec -53.1949 -scale
0.15, the debug shows the geometries of the PSE points and the intermediate
points, and the correlation is visually clear. But then for some reason,
it's not solving. I'll continue having a look at why.

Cameron

**Comment by EAlexJ** [comment]
I was using fastrometry 1.0.10 and just upgraded to 1.0.12. I actually do have some success with some of the files im working with now! I will report back with my findings and if the results I am getting are actually useful.
